code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -68,8 +68,7 @@ stare
 
 /datum/aiTask/prioritizer/flock/on_tick()
 	if(isdead(holder.owner))
-		holder.enabled = FALSE
-		walk(holder.owner, 0) // to prevent moving when dead
+		holder.disable() // to prevent moving when dead
```

Comment:
maybe this should just be in the parent or the code that calls it, assuming we never want a dead mob to have AI?

Reply:
yeah probably. I'm not actually sure it needs to  be here though tbh. I will investigate.